[BE][Attention] Use `isneginf`

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139791
* __->__ #139763
* #139784
* #139788

May be I'm missing some vital piece of information, but it feels like
```c++
  const auto neg_inf = at::scalar_tensor(-std::numeric_limits<float>::infinity(), at::TensorOptions().dtype(out.dtype()).device(out.device()));
  const auto masked = self.eq(neg_inf);
```
should be equivalent to [`torch.isneginf`](https://pytorch.org/docs/stable/generated/torch.isneginf.html) call

malfet committed:
[BE][Transformers] Use `isneginf`

May be I'm missing some vital piece of information, but it feels like
```c++
  const auto neg_inf = at::scalar_tensor(-std::numeric_limits<float>::infinity(), at::TensorOptions().dtype(out.dtype()).device(out.device()));
  const auto masked = self.eq(neg_inf);
```
should be equivalent to [`torch.isneginf`](https://pytorch.org/docs/stable/generated/torch.isneginf.html) call

[ghstack-poisoned]

Skylion007 commented:
@malfet Seems like we need a NestedTensor shim for isinf?

drisspg commented:
+1 @Skylion007 on needing NST support.  Pretty straightforward to add for the pointwise ops

malfet committed:
Update on "[BE][Attention] Use `isneginf`"


May be I'm missing some vital piece of information, but it feels like
```c++
  const auto neg_inf = at::scalar_tensor(-std::numeric_limits<float>::infinity(), at::TensorOptions().dtype(out.dtype()).device(out.device()));
  const auto masked = self.eq(neg_inf);
```
should be equivalent to [`torch.isneginf`](https://pytorch.org/docs/stable/generated/torch.isneginf.html) call

[ghstack-poisoned]

malfet committed:
Update on "[BE][Attention] Use `isneginf`"


May be I'm missing some vital piece of information, but it feels like
```c++
  const auto neg_inf = at::scalar_tensor(-std::numeric_limits<float>::infinity(), at::TensorOptions().dtype(out.dtype()).device(out.device()));
  const auto masked = self.eq(neg_inf);
```
should be equivalent to [`torch.isneginf`](https://pytorch.org/docs/stable/generated/torch.isneginf.html) call

[ghstack-poisoned]

malfet committed:
Update on "[BE][Attention] Use `isneginf`"


May be I'm missing some vital piece of information, but it feels like
```c++
  const auto neg_inf = at::scalar_tensor(-std::numeric_limits<float>::infinity(), at::TensorOptions().dtype(out.dtype()).device(out.device()));
  const auto masked = self.eq(neg_inf);
```
should be equivalent to [`torch.isneginf`](https://pytorch.org/docs/stable/generated/torch.isneginf.html) call

[ghstack-poisoned]

Skylion007 commented on aten/src/ATen/native/nested/NestedTensorUnaryOps.cpp:
```diff
@@ -89,6 +89,10 @@ Tensor NestedTensor_logical_not(const Tensor& self) {
   return map_nt(self, at::logical_not);
 }
 
+Tensor NestedTensor_isneginf(const Tensor& self) {
```
nit: can we add isinf and the other missing functions while wait?

malfet commented:
@pytorchbot merge
